app/ui/client/views/app/room.js

```diff
@@ -799,6 +799,29 @@ Template.room.events({
 		}
 
 		const { username } = msg.u;
+		const roomData = Session.get(`roomData${ RoomManager.openedRoom }`);
```

Comment:
I don't think this is a better implementation of this issue.
My suggestion is to create a custom method on the [roomTypes](https://github.com/RocketChat/Rocket.Chat/blob/develop/app/utils/client/lib/roomTypes.js#L8) class, called by something like `openCustomProfileTab`. In the default method, you would return false and in the `LivechatRoomType` you would implement the method, passing the instance, roomData and username as parameters. Then, you just need to check if the username is the same as the guest username, if so you could do something we do [here](https://github.com/RocketChat/Rocket.Chat/blob/develop/app/livechat/client/hooks/onRenderRoom.js#L12);